### PR TITLE
fix: [PL-39358]: add proper account id filter

### PR DESCRIPTION
**Pull request by xingchi-jin**
## Describe your changes
cherry-pick from https://github.com/harness/harness-core/pull/49570 

Test:
This screenshot below shows that it fixes the issue where some connectors doesn't have logs return
image

Original UI. putting the same params into postman, and the above screenshot shows logs are fetched with this fix
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50099)

### CODE REVIEW

File: 420-delegate-service/src/main/java/io/harness/service/impl/stackdriver/QueryConstructor.java [hunk 1]
```diff
@@ -17,12 +17,12 @@ public class QueryConstructor {
...
       + "jsonPayload.harness.taskId=(%s)\n"
-      + "jsonPayload.harness.accountId=\"%s\"";
+      + "jsonPayload.harness.accountId=\"%s\" OR labels.accountId=\"%s\"";
 
   public static String getTasksLogQuery(String accountId, List<String> taskIds, long start, long end) {
     final List<String> taskIdsWithQuotes =
         taskIds.stream().map(id -> String.format("\"%s\"", id)).collect(Collectors.toList());
     return String.format(TASKS_LOG_QUERY, EpochToUTCConverter.fromEpoch(start), EpochToUTCConverter.fromEpoch(end),
-        StringUtils.join(taskIdsWithQuotes, " OR "), accountId);
+        StringUtils.join(taskIdsWithQuotes, " OR "), accountId, accountId);
```
Comment (raghuAtWings) [inline 420-delegate-service/src/main/java/io/harness/service/impl/stackdriver/QueryConstructor.java:26]: shouldn't this be "AND" instead of "OR"

**Review by raghuAtWings — CHANGES_REQUESTED** [review]

File: 420-delegate-service/src/main/java/io/harness/service/impl/stackdriver/QueryConstructor.java (hunk 1 above)
Comment (xingchi-jin) [reply 420-delegate-service/src/main/java/io/harness/service/impl/stackdriver/QueryConstructor.java:26]: no, our log context is actually using labels.accountId=\"%s\"   
but some places using jsonPayload.harness.accountId=\

only one of them are set in a log line 

i feel OR is safer.

**Comment by xingchi-jin** [comment]
trigger messagecheck

**Comment by xingchi-jin** [comment]
trigger uts

**Comment by xingchi-jin** [comment]
trigger messagecheck

**Comment by xingchi-jin** [comment]
trigger uts

**Pull request merged**